tests/components/group/test_siren.py

```diff
@@ -0,0 +1,432 @@
+    assert hass.states.get("siren.siren_group").state == STATE_UNAVAILABLE
+
+    # At least one member unknown or unavailable -> group unknown
+    hass.states.async_set("siren.test1", STATE_ON)
+    hass.states.async_set("siren.test2", STATE_UNAVAILABLE)
+    await hass.async_block_till_done()
+    assert hass.states.get("siren.siren_group").state == STATE_UNKNOWN
+
+    hass.states.async_set("siren.test1", STATE_ON)
+    hass.states.async_set("siren.test2", STATE_UNKNOWN)
+    await hass.async_block_till_done()
+    assert hass.states.get("siren.siren_group").state == STATE_UNKNOWN
+
+    hass.states.async_set("siren.test1", STATE_UNKNOWN)
+    hass.states.async_set("siren.test2", STATE_UNKNOWN)
+    await hass.async_block_till_done()
+    assert hass.states.get("siren.siren_group").state == STATE_UNKNOWN
+
+    hass.states.async_set("siren.test1", STATE_OFF)
+    hass.states.async_set("siren.test2", STATE_UNAVAILABLE)
+    await hass.async_block_till_done()
+    assert hass.states.get("siren.siren_group").state == STATE_UNKNOWN
+
+    hass.states.async_set("siren.test1", STATE_OFF)
+    hass.states.async_set("siren.test2", STATE_UNKNOWN)
+    await hass.async_block_till_done()
+    assert hass.states.get("siren.siren_group").state == STATE_UNKNOWN
+
+    hass.states.async_set("siren.test1", STATE_UNKNOWN)
+    hass.states.async_set("siren.test2", STATE_UNAVAILABLE)
+    await hass.async_block_till_done()
+    assert hass.states.get("siren.siren_group").state == STATE_UNKNOWN
+
+    # At least one member off -> group off
+    hass.states.async_set("siren.test1", STATE_ON)
+    hass.states.async_set("siren.test2", STATE_OFF)
+    await hass.async_block_till_done()
+    assert hass.states.get("siren.siren_group").state == STATE_OFF
+
+    hass.states.async_set("siren.test1", STATE_OFF)
+    hass.states.async_set("siren.test2", STATE_OFF)
+    await hass.async_block_till_done()
+    assert hass.states.get("siren.siren_group").state == STATE_OFF
+
+    # Otherwise -> on
+    hass.states.async_set("siren.test1", STATE_ON)
+    hass.states.async_set("siren.test2", STATE_ON)
+    await hass.async_block_till_done()
+    assert hass.states.get("siren.siren_group").state == STATE_ON
+
+    # All group members removed from the state machine -> unavailable
+    hass.states.async_remove("siren.test1")
+    hass.states.async_remove("siren.test2")
+    await hass.async_block_till_done()
+    assert hass.states.get("siren.siren_group").state == STATE_UNAVAILABLE
+
+
+@pytest.mark.usefixtures("enable_custom_integrations")
+async def test_service_calls(hass: HomeAssistant) -> None:
+    """Test service calls for sirens."""
+    await async_setup_component(
+        hass,
+        SIREN_DOMAIN,
+        {
+            SIREN_DOMAIN: [
+                {
+                    "platform": DOMAIN,
+                    "entities": ["siren.alarm", "siren.doorbell"],
+                    "all": "false",
+                },
+            ]
```

Comment:
```suggestion
            SIREN_DOMAIN: [
                {"platform": "demo"},
                {
                    "platform": DOMAIN,
                    "entities": ["siren.siren", "siren.siren_with_all_features"],
                    "all": "false",
                },
            ]
```

You can use the `demo` component here, no need to use custom statest